Make sure oauth requests do not timeout during read

regisb opened the pull request:
The previous default timeout of 1s was causing issues in communication
between services, in particular when the server was under load. This
timeout is configurable by passing a `timeout=...` keyword argument to
the OAuthClient constructor; but that requires patching every IDA. It
makes more sense to define a confortable timeout that can be overridden
individually by different IDA.

See these conversations:
https://github.com/edx/edx-rest-api-client/pull/62#issuecomment-646526313
https://discuss.overhang.io/t/error-during-new-setup-no-module-named-ecommerce-settings-tutor/681/9

edx-webhook commented:
Thanks for the pull request, @regisb! I've created [OSPR-4744](https://openedx.atlassian.net/browse/OSPR-4744) to keep track of it in JIRA, where we prioritize reviews.  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

Please let us know once your PR is ready for our review and all tests are green.

regisb commented:
This is ready for review. cc @robrap

natabene commented:
@regisb Thank you for your contribution.

robrap commented:
@regisb: Nit: Your commit comment still has a typo that I had fixed: `confortable`.  Also, looks like this is out-of-date.
Feel free to update, or I'll merge tomorrow.  Can you merge now?

regisb commented:
@robrap I updated the commit message and rebased my changes. I will be away from keyboard until Sunday evening, so I won't be able to make any further changes until then.

regisb commented:
@robrap Do you think you could merge this?

robrap commented:
Sorry @regisb.  This was just forgotten.  Merging away.

edx-webhook commented:
@regisb 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-rest-api-client&entry.867055334=https://github.com/edx/edx-rest-api-client/pull/86&entry.1484655318=https://github.com/regisb&entry.752974735=2020-06-22+18:57&entry.1917517419=2020-08-19+21:44&entry.2133058324=Yes) so we can improve your experience in the future.

robrap commented:
@regisb: I just merged, but then realized I don't think we changed version or added to Changelog. :)  Feel free to ping me if you want to do this and I can merge that too.
